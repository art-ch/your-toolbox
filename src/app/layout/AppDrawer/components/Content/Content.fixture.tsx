import { LuCalculator } from 'react-icons/lu';
import { MdMiscellaneousServices } from 'react-icons/md';

export const APP_DRAWER_CONTENT = [
  {
    icon: <LuCalculator />,
    title: 'Calculators',
    url: '/calculators',
    items: [
      {
        title: 'Ascetic Exercise Duration',
        url: '/calculators/ascetic-exercise-duration'
      }
    ]
  },
  {
    icon: <MdMiscellaneousServices />,
    title: 'Specialized Tools',
    url: '/specialized-tools',
    items: [
      {
        title: 'Rail Signalling Info',
        url: 'https://rail-signalling-info.netlify.app/'
      }
    ]
  }
];
