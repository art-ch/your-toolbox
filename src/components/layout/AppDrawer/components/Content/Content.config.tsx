import { LuCalculator } from 'react-icons/lu';
import { MdMiscellaneousServices } from 'react-icons/md';

export const APP_DRAWER_CONTENT = [
  {
    icon: <LuCalculator />,
    translationKey: 'common:calculators',
    url: '/calculators',
    items: [
      {
        title: 'Ascetic Exercise Duration',
        url: '/calculators/ascetic-exercise-duration'
      }
    ]
  },
  {
    icon: <MdMiscellaneousServices />,
    translationKey: 'common:specializedTools',
    url: '/specialized-tools',
    items: [
      {
        title: 'Rail Signalling Info',
        url: 'https://rail-signalling-info.netlify.app/'
      }
    ]
  }
];
