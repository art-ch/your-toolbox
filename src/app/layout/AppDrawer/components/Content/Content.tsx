import React from 'react';

import {
  SidebarContent,
  SidebarGroup,
  SidebarMenu,
  SidebarMenuButton,
  SidebarMenuItem,
  SidebarMenuSub,
  SidebarMenuSubButton,
  SidebarMenuSubItem,
  SidebarProvider
} from '@/components/ui/sidebar';
import { APP_DRAWER_CONTENT } from './Content.fixture';

export const Content = () => {
  return (
    <SidebarProvider className="min-h-0">
      <SidebarContent>
        <SidebarGroup>
          <SidebarMenu>
            {APP_DRAWER_CONTENT.map((item) => (
              <SidebarMenuItem key={item.title}>
                <SidebarMenuButton asChild>
                  <a href={item.url} className="font-medium">
                    {item?.icon}
                    {item.title}
                  </a>
                </SidebarMenuButton>
                {item.items?.length ? (
                  <SidebarMenuSub className="ml-0 border-l-0 px-1.5">
                    {item.items.map((item) => (
                      <SidebarMenuSubItem key={item.title}>
                        <SidebarMenuSubButton asChild>
                          <a href={item.url}>{item.title}</a>
                        </SidebarMenuSubButton>
                      </SidebarMenuSubItem>
                    ))}
                  </SidebarMenuSub>
                ) : null}
              </SidebarMenuItem>
            ))}
          </SidebarMenu>
        </SidebarGroup>
      </SidebarContent>
    </SidebarProvider>
  );
};
