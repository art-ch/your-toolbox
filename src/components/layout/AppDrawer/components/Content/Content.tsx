'use client';

import React from 'react';

import {
  SidebarContent,
  SidebarGroup,
  SidebarMenu,
  SidebarMenuButton,
  SidebarMenuItem,
  SidebarMenuSub,
  SidebarMenuSubButton,
  SidebarMenuSubItem,
  SidebarProvider
} from '@/components/ui/sidebar';
import { APP_DRAWER_CONTENT } from './Content.config';
import { useTranslation } from 'react-i18next';

export const Content = () => {
  const { t } = useTranslation();

  return (
    <SidebarProvider className="min-h-0">
      <SidebarContent>
        <SidebarGroup>
          <SidebarMenu>
            {APP_DRAWER_CONTENT.map((item) => (
              <SidebarMenuItem key={item.translationKey}>
                <SidebarMenuButton asChild>
                  <a href={item.url} className="font-medium">
                    {item?.icon}
                    {t(item.translationKey)}
                  </a>
                </SidebarMenuButton>
                {item.items?.length ? (
                  <SidebarMenuSub className="ml-0 border-l-0 px-1.5">
                    {item.items.map((item) => (
                      <SidebarMenuSubItem key={item.title}>
                        <SidebarMenuSubButton asChild>
                          <a href={item.url}>{item.title}</a>
                        </SidebarMenuSubButton>
                      </SidebarMenuSubItem>
                    ))}
                  </SidebarMenuSub>
                ) : null}
              </SidebarMenuItem>
            ))}
          </SidebarMenu>
        </SidebarGroup>
      </SidebarContent>
    </SidebarProvider>
  );
};
