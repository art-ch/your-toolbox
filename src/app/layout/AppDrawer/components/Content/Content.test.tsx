import { render } from '@testing-library/react';
import { Content } from './Content';

jest.mock('./Content.fixture', () => ({
  APP_DRAWER_CONTENT: [
    {
      title: 'Item 1',
      url: '/item1',
      icon: null,
      items: []
    },
    {
      title: 'Item 2',
      url: '/item2',
      icon: null,
      items: [
        { title: 'Subitem 1', url: '/subitem1' },
        { title: 'Subitem 2', url: '/subitem2' }
      ]
    }
  ]
}));

jest.mock('@/hooks/use-mobile', () => ({
  useIsMobile: jest.fn()
}));

describe('Content component', () => {
  it('should render correctly', () => {
    const { container } = render(<Content />);

    expect(container).toMatchSnapshot();
  });
});
