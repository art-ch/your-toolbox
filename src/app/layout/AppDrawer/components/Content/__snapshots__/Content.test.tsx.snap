// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content component should render correctly 1`] = `
<div>
  <div
    class="group/sidebar-wrapper flex w-full has-data-[variant=inset]:bg-sidebar min-h-0"
    style="--sidebar-width: 16rem; --sidebar-width-icon: 3rem;"
  >
    <div
      class="flex min-h-0 flex-1 flex-col gap-2 overflow-auto group-data-[collapsible=icon]:overflow-hidden"
      data-sidebar="content"
    >
      <div
        class="relative flex w-full min-w-0 flex-col p-2"
        data-sidebar="group"
      >
        <ul
          class="flex w-full min-w-0 flex-col gap-1"
          data-sidebar="menu"
        >
          <li
            class="group/menu-item relative"
            data-sidebar="menu-item"
          >
            <a
              class="peer/menu-button flex w-full items-center gap-2 overflow-hidden rounded-md p-2 text-left outline-hidden ring-sidebar-ring transition-[width,height,padding] focus-visible:ring-2 active:bg-sidebar-accent active:text-sidebar-accent-foreground disabled:pointer-events-none disabled:opacity-50 group-has-data-[sidebar=menu-action]/menu-item:pr-8 aria-disabled:pointer-events-none aria-disabled:opacity-50 data-[active=true]:bg-sidebar-accent data-[active=true]:font-medium data-[active=true]:text-sidebar-accent-foreground data-[state=open]:hover:bg-sidebar-accent data-[state=open]:hover:text-sidebar-accent-foreground group-data-[collapsible=icon]:size-8! group-data-[collapsible=icon]:p-2! [&>span:last-child]:truncate [&>svg]:size-4 [&>svg]:shrink-0 hover:bg-sidebar-accent hover:text-sidebar-accent-foreground h-8 text-sm font-medium"
              data-active="false"
              data-sidebar="menu-button"
              data-size="default"
              href="/item1"
            >
              Item 1
            </a>
          </li>
          <li
            class="group/menu-item relative"
            data-sidebar="menu-item"
          >
            <a
              class="peer/menu-button flex w-full items-center gap-2 overflow-hidden rounded-md p-2 text-left outline-hidden ring-sidebar-ring transition-[width,height,padding] focus-visible:ring-2 active:bg-sidebar-accent active:text-sidebar-accent-foreground disabled:pointer-events-none disabled:opacity-50 group-has-data-[sidebar=menu-action]/menu-item:pr-8 aria-disabled:pointer-events-none aria-disabled:opacity-50 data-[active=true]:bg-sidebar-accent data-[active=true]:font-medium data-[active=true]:text-sidebar-accent-foreground data-[state=open]:hover:bg-sidebar-accent data-[state=open]:hover:text-sidebar-accent-foreground group-data-[collapsible=icon]:size-8! group-data-[collapsible=icon]:p-2! [&>span:last-child]:truncate [&>svg]:size-4 [&>svg]:shrink-0 hover:bg-sidebar-accent hover:text-sidebar-accent-foreground h-8 text-sm font-medium"
              data-active="false"
              data-sidebar="menu-button"
              data-size="default"
              href="/item2"
            >
              Item 2
            </a>
            <ul
              class="mx-3.5 flex min-w-0 translate-x-px flex-col gap-1 border-sidebar-border py-0.5 group-data-[collapsible=icon]:hidden ml-0 border-l-0 px-1.5"
              data-sidebar="menu-sub"
            >
              <li>
                <a
                  class="flex h-7 min-w-0 -translate-x-px items-center gap-2 overflow-hidden rounded-md px-2 text-sidebar-foreground outline-hidden ring-sidebar-ring hover:bg-sidebar-accent hover:text-sidebar-accent-foreground focus-visible:ring-2 active:bg-sidebar-accent active:text-sidebar-accent-foreground disabled:pointer-events-none disabled:opacity-50 aria-disabled:pointer-events-none aria-disabled:opacity-50 [&>span:last-child]:truncate [&>svg]:size-4 [&>svg]:shrink-0 [&>svg]:text-sidebar-accent-foreground data-[active=true]:bg-sidebar-accent data-[active=true]:text-sidebar-accent-foreground text-sm group-data-[collapsible=icon]:hidden"
                  data-sidebar="menu-sub-button"
                  data-size="md"
                  href="/subitem1"
                >
                  Subitem 1
                </a>
              </li>
              <li>
                <a
                  class="flex h-7 min-w-0 -translate-x-px items-center gap-2 overflow-hidden rounded-md px-2 text-sidebar-foreground outline-hidden ring-sidebar-ring hover:bg-sidebar-accent hover:text-sidebar-accent-foreground focus-visible:ring-2 active:bg-sidebar-accent active:text-sidebar-accent-foreground disabled:pointer-events-none disabled:opacity-50 aria-disabled:pointer-events-none aria-disabled:opacity-50 [&>span:last-child]:truncate [&>svg]:size-4 [&>svg]:shrink-0 [&>svg]:text-sidebar-accent-foreground data-[active=true]:bg-sidebar-accent data-[active=true]:text-sidebar-accent-foreground text-sm group-data-[collapsible=icon]:hidden"
                  data-sidebar="menu-sub-button"
                  data-size="md"
                  href="/subitem2"
                >
                  Subitem 2
                </a>
              </li>
            </ul>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
